src/PixiEditor/Views/Overlays/TransformOverlay/TransformUpdateHelper.cs

```diff
@@ -68,23 +68,15 @@ public static ShapeCorners? UpdateShapeFromCorner
             VecD leftNeighborPos = TransformHelper.GetAnchorPosition(corners, leftNeighbor);
             VecD rightNeighborPos = TransformHelper.GetAnchorPosition(corners, rightNeighbor);
 
-            double angle = corners.RectRotation;
-            if (double.IsNaN(angle))
-                angle = 0;
-
-            if (scaleFromCenter)
-            {
-                return ScaleCornersFromCenter(corners, targetCorner, desiredPos, angle);
-            }
-
-            // find positions of neighboring corners relative to the opposite corner, while also undoing the transform rotation
-            VecD targetTrans = (targetPos - oppositePos).Rotate(-angle);
-            VecD leftNeighTrans = (leftNeighborPos - oppositePos).Rotate(-angle);
-            VecD rightNeighTrans = (rightNeighborPos - oppositePos).Rotate(-angle);
+            // find positions of neighboring corners relative to the opposite corner
+            VecD targetTrans = targetPos - oppositePos;
+            VecD leftNeighTrans = leftNeighborPos - oppositePos;
+            VecD rightNeighTrans = rightNeighborPos - oppositePos;
 
             // find by how much move each corner
-            VecD delta = (desiredPos - targetPos).Rotate(-angle);
+            VecD delta = desiredPos - targetPos;
             VecD leftNeighDelta, rightNeighDelta;
+            VecD oppositeDelta = scaleFromCenter ? -delta : VecD.Zero;
```

Comment:
If scaling from center, move the opposite corner the same amount as the held corner